client/lib/protect-form/index.tsx

```diff
@@ -108,7 +103,7 @@ export class ProtectFormGuard extends Component {
 		markSaved( this );
 	}
 
-	componentWillReceiveProps( nextProps ) {
+	componentWillReceiveProps( nextProps: ProtectFormGuardProps ) {
```

Comment:
I was surprised I needed to type this. I thought it would be inferred by React Component types?